cmd/gardenlet/app/migration.go

```diff
@@ -165,3 +179,85 @@ func syncBackupSecretRefAndCredentialsRef(backup *gardencorev1beta1.SeedBackup)
 	// - both fields are set -> let the validation check if they are correct
 	// - credentialsRef refer to WorkloadIdentity -> secretRef should stay unset
 }
+
+func migrateMCMRBAC(ctx context.Context, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
+		return err
+	}
+
+	var tasks []flow.TaskFn
+
+	for _, ns := range namespaceList.Items {
+		if ns.DeletionTimestamp != nil || ns.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+		namespace := ns
+		tasks = append(tasks, func(ctx context.Context) error {
+			clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			if err := seedClient.Get(ctx, client.ObjectKey{Name: "machine-controller-manager-" + namespace.Name}, clusterRoleBinding); err != nil {
+				//If MCM clusterRoleBinding does not exist, nothing to do
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
```

Comment:
Why do you return here? If this succeeds for a namespace but fails in the next step (when `mcm.DeployMigrate` is called, another invocation of this entire function will not try to create the new RBAC roles since it exists early here.

Reply:
`mcm.DeployMigrate` is ordered in a way that it creates the RBAC roles before the `clusterRoleBinding` is deleted. I think it's fair to return here since we would know that RBAC have been created